packages/contracts/contracts/core/DAO.sol

```diff
@@ -215,47 +248,156 @@ contract DAO is
     /// @inheritdoc IDAO
     function deposit(
         address _token,
+        uint256 _tokenId,
         uint256 _amount,
         string calldata _reference
-    ) external payable override {
-        if (_amount == 0) revert ZeroAmount();
-
+    ) external payable {
+        // -- Native Token -- //
         if (_token == address(0)) {
-            if (msg.value != _amount)
+            if (_amount == 0) {
+                revert ZeroAmount();
+            }
+            if (_tokenId != 0) {
+                revert NonZeroTokenId(_tokenId);
+            }
+            if (msg.value != _amount) {
                 revert NativeTokenDepositAmountMismatch({expected: _amount, actual: msg.value});
+            }
         } else {
-            if (msg.value != 0)
+            if (msg.value != 0) {
                 revert NativeTokenDepositAmountMismatch({expected: 0, actual: msg.value});
+            }
 
-            IERC20Upgradeable(_token).safeTransferFrom(msg.sender, address(this), _amount);
+            // -- ERC-165 Supporting Tokens -- //
+            if (_token.supportsERC165()) {
```

Comment:
This is not correct.  `supportsInterface` on the below line anyways calls ERC165 check.

Reply:
But it could be that it is an `ERC-20` which doesn't support `ERC-165` so that the call would revert, no?

Reply:
no, check OZ's supportsInterface and how it does. It includes the check and you're doing it again.